superset/utils/geocoding_utils.py

```diff
@@ -0,0 +1,136 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import json
+import time
+
+import requests
+from requests import RequestException
+
+
+class GeoCoder:  # pylint: disable=too-few-public-methods
+    """
+    The GeoCoder object holds the logic for geocoding given data
+    """
+
+    interruptflag = False
+    conf: dict = {}
+    progress: dict = {}
+
+    def __init__(self, config):
+        self.conf = config
+
+    def geocode(self, geocoder: str, data: list, async_flag=False):
+        try:
+            if geocoder == "MapTiler":
+                return self.__geocode_maptiler(data, async_flag)
+            return self.__geocode_testing(async_flag)
+        except Exception as e:
+            raise e
+        finally:
+            self.progress["progress"] = 0
+            self.progress["is_in_progress"] = False
+
+    def __geocode_maptiler(self, data: list, async_flag) -> list:
```

Comment:
Add type hint for async_flag (bool) or remove it entirely as this is not used and will not be enabled at least not as part of this project